tests/Integration/Behaviour/Features/Context/CarrierFeatureContext.php

```diff
@@ -0,0 +1,375 @@
+        $carrier->active = 1;
+        $carrier->add();
+        $this->carriers[$carrierName] = $carrier;
+
+        $groups = Group::getGroups(Context::getContext()->language->id);
+        $groupIds = [];
+        foreach ($groups as $group) {
+            $groupIds[] = $group['id_group'];
+        }
+        $carrier->setGroups($groupIds);
+    }
+
+    /**
+     * @Given /^Carrier with name (.+) ships to all groups$/
+     */
+    public function setCarrierToAllGroups($carrierName)
+    {
+        $this->checkCarrierWithNameExists($carrierName);
+        $carrier = $this->carriers[$carrierName];
+
+        $groups = Group::getGroups(Context::getContext()->language->id);
+        $groupIds = [];
+        foreach ($groups as $group) {
+            $groupIds[] = $group['id_group'];
+        }
+        $carrier->setGroups($groupIds);
+    }
+
+    /**
+     * @param $carrierName
+     */
+    public function checkCarrierWithNameExists($carrierName)
+    {
+        if (!isset($this->carriers[$carrierName])) {
+            throw new \Exception('Carrier with name "' . $carrierName . '" was not added in fixtures');
+        }
+    }
+
+    /**
+     * @param $carrierName
+     *
+     * @return Carrier
+     */
+    public function getCarrierWithName($carrierName)
+    {
+        return $this->carriers[$carrierName];
+    }
+
+    /**
+     * Be careful: this method REPLACES shipping fees for carrier
+     *
+     * @Given /^Carrier with name (.+) has a shipping fees of (\d+\.\d+) in zone with name (.+) for quantities between (\d+) and (\d+)$/
+     */
+    public function setCarrierFees($carrierName, $shippingPrice, $zoneName, $fromQuantity, $toQuantity)
+    {
+        $this->checkCarrierWithNameExists($carrierName);
+        $this->checkZoneWithNameExists($zoneName);
+        if (empty($this->carriers[$carrierName]->getZone((int) $this->zones[$zoneName]->id))) {
+            $this->carriers[$carrierName]->addZone((int) $this->zones[$zoneName]->id);
+        }
+        $rangeId = RangePrice::rangeExist($this->carriers[$carrierName]->id, $fromQuantity, $toQuantity);
+        if (!empty($rangeId)) {
+            $range = new RangePrice($rangeId);
+        } else {
+            $range = new RangePrice();
+            $range->id_carrier = $this->carriers[$carrierName]->id;
+            $range->delimiter1 = $fromQuantity;
+            $range->delimiter2 = $toQuantity;
+            $range->add();
+            $this->priceRanges[] = $range;
+        }
+        $carrierPriceRange = [
+            'id_range_price' => (int) $range->id,
+            'id_range_weight' => null,
+            'id_carrier' => (int) $this->carriers[$carrierName]->id,
+            'id_zone' => (int) $this->zones[$zoneName]->id,
+            'price' => $shippingPrice,
+        ];
+        $this->carriers[$carrierName]->addDeliveryPrice([$carrierPriceRange], true);
+    }
+
+    /**
+     * @AfterScenario
+     */
+    public function cleanData()
```

Comment:
Data is a bit vague, isn't it rather `cleanFixtures` ? or `emptyFixturesFromDatabase` ?

If the real name is too long, maybe add a phpDoc description ? The idea is to help other devs writing behat steps to see what they can reuse and what they need to add.